src/backends/sqlite3/statement.cpp

```diff
@@ -343,13 +344,13 @@ sqlite3_statement_backend::execute(int number)
     }
     else
     {
-        if (1 == number)
+        if (hasVectorIntoElements_ || number == 0)
         {
-            retVal = load_one();
+            retVal = load_rowset(number);
         }
         else
         {
-            retVal = load_rowset(number);
+            retVal = load_one();
         }
```

Comment:
Shouldn't we just call `fetch()` here if the code is identical in both places now?